include/processor_cxx17/msg_handler.hxx

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "base_message.hxx"
+#include "tcp_server_connection.hxx"
+
+#include "control_message.hxx"
+#include "control_message_ack.hxx"
+#include "control_message_ack_with_checksum.hxx"
+#include "control_message_nack.hxx"
+#include "data_message.hxx"
+
+#include <memory>
+#include <boost/asio.hpp>
+#include <variant>
+
+namespace cpp17_app {
+
+class MessageHandler: public net::TcpServerConnection {
+public:
+    explicit MessageHandler(boost::asio::io_context &io_context);
+    ~MessageHandler() = default;
+
+    MessageHandler(const MessageHandler&) = delete;
+    MessageHandler& operator=(const MessageHandler&) = delete;
+
+    MessageHandler(MessageHandler&&) = delete;
+    MessageHandler& operator=(MessageHandler&&) = delete;
+
+    using container_t = std::vector<std::byte>;
+
+    using MessageVariant = std::variant<
```

Comment:
IMHO this should be private/protected since this type is relevant only for the class hierarchy.